### PR TITLE
Add support for API queries in CIS-CAT and Syscollector

### PR DESCRIPTION
**Pull request by mgmacias95**
Hello team,

This PR adds support for API queries in Syscollector and CIS-CAT endpoints. It also adds support to use `wazuh-db` as database backend in the framework and will help us in the future, when SQLite based requests go deprecated.

Best regards,
Marta

### CODE REVIEW

File: framework/wazuh/utils.py [hunk 1]
```diff
@@ -613,7 +614,8 @@ class WazuhDBQuery(object):
     This class describes a database query for wazuh
     """
     def __init__(self, offset, limit, table, sort, search, select, query, fields, default_sort_field, db_path, count,
-                 get_data, default_sort_order='ASC', filters={}, min_select_fields=set(), date_fields=set(), extra_fields=set()):
+                 get_data, default_sort_order='ASC', filters={}, min_select_fields=set(), date_fields=set(),
+                 extra_fields=set(), backend='sqlite3', agent_id=None):
```
Comment (crd1985) [inline framework/wazuh/utils.py]: The `backend` param could be redesign in a smarter way. Maybe we could isolate the lines in this code where you make a conditional based on the backend into separate methods of a new class `DBBackend`. Then you could build a `WazuhDBBackend` and a `Sqlite3Backend` classes. Each one of these classes would implement the specific logic to deal with different databases o services (like wazuh-db).

Comment (mgmacias95) [reply framework/wazuh/utils.py]: Hello @crd1985,

I have added new commits including the suggested refactor. All unit tests are correct:
```python
# /var/ossec/framework/python/bin/pytest wazuh/tests/test_ciscat.py
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 1 item

wazuh/tests/test_ciscat.py .                                                                                                                                                                                                                                          [100%]

========================================================================================================================= 1 passed in 0.25 seconds ==========================================================================================================================
# /var/ossec/framework/python/bin/pytest wazuh/tests/test_syscollector.py
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 16 items

wazuh/tests/test_syscollector.py ................                                                                                                                                                                                                                     [100%]

========================================================================================================================= 16 passed in 0.36 seconds =========================================================================================================================
# /var/ossec/framework/python/bin/pytest wazuh/tests/test_security_configuration_assessment.py
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 4 items

wazuh/tests/test_security_configuration_assessment.py ....                                                                                                                                                                                                            [100%]

========================================================================================================================= 4 passed in 0.27 seconds ==========================================================================================================================
# /var/ossec/framework/python/bin/pytest wazuh/tests/test_rootcheck.py
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 9 items

wazuh/tests/test_rootcheck.py .........                                                                                                                                                                                                                               [100%]

========================================================================================================================= 9 passed in 0.31 seconds ==========================================================================================================================
# /var/ossec/framework/python/bin/pytest wazuh/tests/test_agent.py
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 39 items

wazuh/tests/test_agent.py .......................................                                                                                                                                                                                                     [100%]

========================================================================================================================= 39 passed in 0.45 seconds =========================================================================================================================
```

File: framework/wazuh/utils.py [hunk 2]
```diff
@@ -671,26 +735,21 @@ def __init__(self, offset, limit, table, sort, search, select, query, fields, de
...
-        self.conn = Connection(db_path)
-
+        self.backend = backend
 
     def _add_limit_to_query(self):
         if self.limit:
             if self.limit > common.maximum_database_limit:
                 raise WazuhException(1405, str(self.limit))
-            self.query += ' LIMIT :offset,:limit'
+            self.query += f' LIMIT :limit OFFSET :offset'
```
Comment (crd1985) [inline framework/wazuh/utils.py]:
```suggestion
            self.query += ' LIMIT :limit OFFSET :offset'
```

File: framework/wazuh/utils.py [hunk 3]
```diff
@@ -841,25 +892,29 @@ def _add_filters_to_query(self):
             self.query += ('))' if curr_level > q_filter['level'] else ')') + ' {} '.format(q_filter['separator'])
             curr_level = q_filter['level']
 
+    def _substitute_params(self):
```
Comment (crd1985) [inline framework/wazuh/utils.py]: This function can be removed

**Comment by crd1985** [comment]
Closed in favor of https://github.com/wazuh/wazuh/pull/3747

**Pull request closed without merging**